src/CommonProviderImplementation/Helpers.fs

```diff
@@ -302,10 +305,11 @@ module internal ProviderHelpers =
                         providedTypesCache.TryRemove key |> ignore
                     else
                         log (sprintf "Saving generation of type %s for 10 seconds awaiting incremental recreation [%d]" fullTypeName tp.Id)
-                        providedTypesCache.[key] <- CacheValue (providedType, fullKey)
+                        providedTypesCache.AddOrUpdate(key, CacheValue (providedType, fullKey), fun _ _ -> CacheValue (providedType, fullKey)) |> ignore
                         // Remove the cache entry in 10 seconds
                         async { do! Async.Sleep (10000)
-                                providedTypesCache.TryRemove(key) |> ignore } |> Async.StartImmediate
+                                if providedTypesCache <> null then
```

Comment:
That's right it shouldn't be. I was getting ObjectDisposedExceptions while I had debugger attached to FSharp.Data and breaking to every exception. So I assumed that some async method like this does try to handle object that is already disposed. But clearly not this one as the dictionary is static.